impeller/aiks/canvas.cc

```diff
@@ -23,6 +23,79 @@
 
 namespace impeller {
 
+namespace {
+
+static std::shared_ptr<Contents> CreateContentsForGeometryWithFilters(
```

Comment:
While this code uses shared_ptr its careful to move when possible to avoid the shared ptr release ops.

Reply:
No action required: rvalue parameters are technically forbidden ( https://google.github.io/styleguide/cppguide.html#Rvalue_references ) but it would have been nice to statically assert that people are moving `geometry` into this.  I see there is just one case were you'd have to manually copy it to get an rvalue.

This is beyond the scope of this PR but we could have created a wrapper for std::shared_ptr that doesn't have a copy constructor.
```c++
template<typename T>
class  move_ptr {
  public:
    move_ptr(const ptr<T>&) = delete;
    move_ptr(const ptr<T>&&) = default;
    move_ptr<T> Clone() const {...}
  private:
   std::shared_ptr<T> ptr_;
};
```